sdl_android/src/main/java/com/smartdevicelink/transport/RouterServiceValidator.java

```diff
@@ -513,7 +513,7 @@ public boolean isAdditionalSdlBroadcastReceiver(){
 			String packageName = resolveInfo.activityInfo.packageName;
 			if(packageName != null){
 				if(lastClosedService != null){
-					if(isTrustedPackage(packageName, manager) && !lastClosedService.equals(packageName)){
+					if(isTrustedPackage(packageName, manager) && !lastClosedService.toString().equals(packageName)){
 						return true;
```

Comment:
This condition check will not work the way it's intent to:
`!lastClosedService.toString().equals(packageName)`

The reason is that `lastClosedService` is a ComponentName and its toString method returns: 
`"ComponentInfo{" + mPackage + "/" + mClass + "}"`
which will never equal to `packageName`.

I think it should be: `!lastClosedService.getPackageName().equals(packageName)` if the intention is to exclude the current (last closed) service.